suppress update

nmercadeb opened the pull request.

catalamarti commented:
hi @nmercadeb I've refactored a bit the suppress function so it is clearer and we split the different reasons for censoring. Could you please take a look, ensure it makes sense?

nmercadeb commented:
> hi @nmercadeb I've refactored a bit the suppress function so it is clearer and we split the different reasons for censoring. Could you please take a look, ensure it makes sense?

Looks good to me!
